docs/devel.md

```diff
@@ -146,6 +146,57 @@ The service log will show:
 2020-06-08 15:12:39,478 127.0.0.1 - - [08/Jun/2020 15:12:39] "GET /repos/app-sre/sretoolbox HTTP/1.1" 200 -
 ```
 
+## Run github-mirror in Redis backed mode (optional)
+
+When running in Redis mode, github-mirror will cache all requests in a Redis server instead of locally in-memory. This enables multiple instances of github-mirror to access and maintain a single shared cache.
+
+### A. Configuring Redis
+First make sure to download and install [Redis](https://redis.io/download).
+
+Since the default Redis configuration does not require authentication, you will also need to create a Redis configuration file.
+
+Create a simple configuration file `/etc/redis/sample_config.conf` with a Redis authentication password:
+
+```
+requirepass <mysecret>
+```
+
+From a different terminal run the Redis server:
+
+```
+user@localhost:~$ redis-server /etc/redis/sample_config.conf
+```
+
+Use the Redis client `redis-cli` to test your connection to the Redis server.
+
+By default, `redis-cli` will connect to the server at `127.0.0.1` port `6379`:
+```
+user@localhost:~$ redis-cli -a <mysecret>
+127.0.0.1:6379> ping
+PONG
+```
+
+### B. Configuring github-mirror to run in Redis mode
+
+First you will need to tell github-mirror to run in Redis mode by setting the `CACHE_TYPE` environment variable. If the variable is not set, github mirror will run in the default in-memory cache mode.
+
+```
+CACHE_TYPE = redis
+```
+
+You will also need to set up the following environment variables:
+```
+REDIS_ENDPOINT = <redis.endpoint>
+REDIS_PORT = <port>
+REDIS_PASSWORD = <mysecret>
+```
+
+where `REDIS_PASSWORD` is the same password set in the Redis configuration file. `REDIS_ENDPOINT` and `REDIS_PORT` are the address and port of the Redis server (typically this is `localhost` and `6379`).
+
+[Please note that github-mirror expects an encrypted SSL connection to Redis, so ensure you are connecting to Redis via an SSL encrypted tunnel or manually disable SSL encryption [here](https://github.com/app-sre/github-mirror/blob/master/ghmirror/data_structures/redis_data_structures.py#L36)].
```

Comment:
I kind of hate this. Ideally I think that ssl encryption should be an environment varaible whose value is a parameter in the saas file and defaults to True. That way when running this locally, I can simply set the environment variable to false without editing the code.

Reply:
Will it work with the simplest redis configuration for testing?

Reply:
you mean like running make check?

Reply:
but yes if I pull the value of ssl from an environment variable, i can test `curl` requests to github mirror

Reply:
Yeah, I mean, you propose in the docs to run redis in a given way. But the source code won't work with that redis. Docs should give some steps that actually work. So, if SSL is a requirement, it's ok to say in the docs "go deploy redis with ssl then come back here" :)
Or, as you said, pull the ssl value from the environment so a developer can have a simple env to play with.

Reply:
great advice as usual, i hope you like the new and improved version